aggregator-microservices/aggregator-service/src/main/java/com/iluwatar/aggregator/microservices/Aggregator.java

```diff
@@ -50,8 +50,21 @@
   @RequestMapping("/product")
   public Product getProduct() {
     Product product = new Product();
-    product.setTitle(informationClient.getProductTitle());
-    product.setProductInventories(inventoryClient.getProductInventories());
+    String productTitle = informationClient.getProductTitle();
+    Integer productInventory = inventoryClient.getProductInventories();
+
+    if (productTitle != null) {
+      product.setTitle(productTitle);
+    } else {
+      product.setTitle("Error: Fetching Product Title Failed"); //Fallback to error message
+    }
+
+    if (productInventory != null) {
```

Comment:
`ProductInventoryClientImpl` does not return null but 0 when exception occurs. Perhaps returning null would be the correct way, so the change should be implemented in the client..